nodeup/pkg/model/containerd.go

```diff
@@ -407,12 +407,17 @@ iptables -w -t nat -A IP-MASQ -d {{.NonMasqueradeCIDR}} -m comment --comment "ip
 iptables -w -t nat -A IP-MASQ -m comment --comment "ip-masq: outbound traffic is subject to MASQUERADE (must be last in chain)" -j MASQUERADE
 `
 
-	if b.Cluster.Spec.Networking.NonMasqueradeCIDR == "" {
-		// We could fall back to the pod CIDR, that is likely more correct anyway
-		return fmt.Errorf("NonMasqueradeCIDR is not set")
+	nonMasqueradeCIDR := b.Cluster.Spec.Networking.NonMasqueradeCIDR
+	if nonMasqueradeCIDR == "" {
+		if podCIDR := b.Cluster.Spec.Networking.PodCIDR; podCIDR != "" {
+			klog.Warningf("defaulting nonMasqueradeCIDR to podCIDR %q", podCIDR)
```

Comment:
Why not populate `NonMasqueradeCIDR` in the completed cluster spec, perhaps in `performNetworkAssignmentsIPAliases()`?

If iptables is disabling masquerading for the CIDR, then it truly is the `NonMasqueradeCIDR`. Why should GCP in particular bind the value so much later than other cloud providers?

Reply:
I would also do this in cloudup. Doing it here would be a bit surprising and I would probably not think to look here.